src/test/java/fi/metatavu/kuntaapi/test/utils/PtvDownload.java

```diff
@@ -20,39 +27,124 @@
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.JsonMappingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 
-import fi.metatavu.ptv.client.model.V5VmOpenApiOrganizationService;
-import fi.metatavu.ptv.client.model.V7VmOpenApiOrganization;
-import fi.metatavu.ptv.client.model.V7VmOpenApiService;
-import fi.metatavu.ptv.client.model.V7VmOpenApiServiceServiceChannel;
 import fi.metatavu.kuntaapi.server.integrations.ptv.servicechannels.ServiceChannelType;
+import fi.metatavu.ptv.client.model.V5VmOpenApiOrganizationService;
+import fi.metatavu.ptv.client.model.V8VmOpenApiOrganization;
+import fi.metatavu.ptv.client.model.V8VmOpenApiService;
+import fi.metatavu.ptv.client.model.V8VmOpenApiServiceServiceChannel;
+import fi.metatavu.ptv.client.model.VmOpenApiCodeListItem;
+import fi.metatavu.ptv.client.model.VmOpenApiItem;
 
 @SuppressWarnings ({"squid:S106", "squid:S3457", "squid:S1148"})
 public class PtvDownload {
+
+  private static final boolean PRINT_IDS = true;
+  private static final boolean DOWNLOAD_CODES = false;
+  private static final boolean DOWNLOAD_RESOURCES = false;
+  private static final boolean PURGE_SERVICES = false;
+  private static final boolean PURGE_SERVICECHANNELS = false;
+  private static final boolean LIMIT_CHANNELS = true;
   
   private static final String JSON = ".json";
-  private static final String PTV = "v7";
+  private static final String PTV = "v8";
+  private static final int POSTAL_CODE_PAGES = 5;
+  private static final int ORGANIZATION_SERVICE_LIMIT = 10;
+  private static final int CHANNEL_LIMIT = 10;
 
   public static void main(String[] args) throws IOException {
-    new PtvDownload()
-      .downloadCodes();
+    PtvDownload instance = new PtvDownload();
+
+    if (PURGE_SERVICES) {
+      instance.purgeServices();
+    }
+
+    if (PURGE_SERVICECHANNELS) {
+      instance.purgeServiceChannels();
+    }
+
+    if (DOWNLOAD_CODES) {
+      instance.downloadCodes();
+    }
+    
+    if (DOWNLOAD_RESOURCES) {
+      instance.downloadOrganizations();
+    }
+    
+    if (LIMIT_CHANNELS) {
+      instance.limitChannels();
+    }
+    
+    if (PRINT_IDS) {
+      instance.printIds();
+    }
+  }
+
+  private void limitChannels() throws JsonParseException, JsonMappingException, IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)